apps/web/src/pages/templates/editor/TourProvider.tsx

```diff
@@ -0,0 +1,89 @@
+import { useMantineColorScheme } from '@mantine/core';
+import React, { useMemo, useCallback } from 'react';
+import Joyride from 'react-joyride';
+import { colors } from '../../../design-system';
+
+import { useDigestWorkflowTour } from './useDigestWorkflowTour';
+
+const TourContext = React.createContext({
+  startTour: () => {},
+  stopTour: () => {},
+  stepTourIndex: 0,
+  setStep: (index) => {},
+});
+
+export const useTour = () => React.useContext(TourContext);
+
+export const TourProvider = React.memo(() => {
```

Comment:
The tour provider is a wrapper around the `react-joyride` **Joyride** component  and controls its state like it has a functionality to start/stop the tour and set the current tour hint step